### PR TITLE
Optional skip param for FileNet_SendFile

### PR DESCRIPTION
**Pull request by Nolo001-Aha**
Lets plugin authors choose whether they want File Network to query sv_allowupload in FileNet_SendFile or not. Can be rather useful in cases where plugins relying on File Network have their own cvar query implementations, and having a 2nd one would complicate the logic. (For example, my solution needs FileNetwork to queue the file as fast as possible without any timers and additional queries involved, I check cvars before I get to send/request anything).

### CODE REVIEW

**Review by Batfoxkid — COMMENTED** [review]
I think I'll be more in for removing the sv_allowupload check itself. Very niche use case of having it in here and would be more appropriate that it'll return a fail when it client rejects it.

**Comment by Nolo001-Aha** [comment]
If I'm not mistaken, even with cl_allowupload set to 0 (I _think_ cl_allowupload is the correct ConVar, as **cl** stands for client and that's what clients can use to toggle uploading custom content to servers; sv_allowupload is the server-side ConVar that acts as a global on-off switch whether clients are allowed to upload stuff or not) clients fully receive the file but then discard it once the file is reconstructed.

Pretty sure it's ok to go straight for Timer_SendingClient then, without the ConVar query. Would you like me to pr that?

**Comment by Batfoxkid** [comment]
> If I'm not mistaken, even with cl_allowupload set to 0 (I _think_ cl_allowupload is the correct ConVar, as **cl** stands for client and that's what clients can use to toggle uploading custom content to servers; sv_allowupload is the server-side ConVar that acts as a global on-off switch whether clients are allowed to upload stuff or not) clients fully receive the file but then discard it once the file is reconstructed.
> 
> Pretty sure it's ok to go straight for Timer_SendingClient then, without the ConVar query. Would you like me to pr that?

Yeah, just skipping the ConVar query will be best.

**Comment by Nolo001-Aha** [comment]
Should be fine now I think

File: scripting/include/filenetwork.inc [hunk 1]
```diff
@@ -84,7 +84,7 @@ native bool FileNet_IsFileInWaitingList(int client, const char[] file);
  * 
  * @return			True if the file is now queued to be sent, false otherwise
  */
-native bool FileNet_SendFile(int client, const char[] file, FileNet_SendFileResult callback = INVALID_FUNCTION, any data = 0);
+native bool FileNet_SendFile(int client, const char[] file, FileNet_SendFileResult callback = INVALID_FUNCTION, any data = 0, bool skip = false);
```
Comment (Batfoxkid) [inline scripting/include/filenetwork.inc]: This change is still present.

**Review by Batfoxkid — CHANGES_REQUESTED** [review]

File: scripting/include/filenetwork.inc (hunk 1 above)
Comment (Nolo001-Aha) [reply scripting/include/filenetwork.inc]: Done, how blind can I get...

**Comment by y0edan** [comment]
Can't seem to get the plugin running and keep getting the following error;

L 04/18/2023 - 00:52:38: [SM] Exception reported: [Gamedata] EngineInterface is incorrect for mod
L 04/18/2023 - 00:52:38: [SM] Blaming: core/filenetwork.smx
L 04/18/2023 - 00:52:38: [SM] Call stack trace:
L 04/18/2023 - 00:52:38: [SM]   [0] SetFailState
L 04/18/2023 - 00:52:38: [SM]   [1] Line 99, plugin.sp::OnPluginStart

CS:GO - 
 DHooks (1.11.0.6930) by AlliedModders LLC: Dynamic Hooks
 SourceMod 1.11.0.6930, by AlliedModders LLC
 
 Any ideas?

**Comment by y0edan** [comment]
Thank you!! Will test tonight.

**Pull request closed without merging**